src/jit/codegenxarch.cpp

```diff
@@ -2470,6 +2470,18 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
     case GT_JTRUE:
         {
             GenTree *cmp = treeNode->gtOp.gtOp1;
+
+#if !defined(_TARGET_64BIT_)
+            GenTreePtr op1 = cmp->gtGetOp1();
+            bool isSigned = ((cmp->gtFlags & GTF_UNSIGNED) == 0);
+
+            // For long signed compares that are not GT_NE or GT_EQ, we emit the logic in genCompareLong,
+            // and do not need to do anything here.
+            if (varTypeIsLong(op1->TypeGet()) && isSigned && (cmp->gtOper != GT_NE && cmp->gtOper != GT_EQ))
```

Comment:
Makes sense. I'll combine the logic for signed and unsigned in genCompareLong, and change the if in JTRUE.